### DISCUSSION TITLE
Component proposal: ChatUI

### DISCUSSION THREAD

**Discussion by jefffortune** [uswds/uswds#5938] [Proposals]
### Component name

- ChatUI
- AI Chat Interface
- Conversational UI
- Conversation box

### Component description

The popularity of conversational AI agents like ChatGPT, Hugging Face Chat, or Anthropic is setting new standards for user interaction. There is a growing need for a human-centered and accessible UI for these conversational AI agents. Currently, we are developing a simplified demo to meet our immediate needs.

The government could benefit from a component or set of components for creating conversational agents using the U.S. Web Design System (USWDS) and ensuring government sites are 508 compliant. A focus on human-centered design will provide an intuitive, engaging, and user-friendly UI, accommodating diverse user needs and preferences. To work for all users of the government, from back-end users to end users who consume or provide information.

### Why USWDS needs this component

To support government initiatives for 508 compliance, ensuring all individuals have equal access and opportunities to interact with AI agents developed by the government. A human-centered approach ensures the interface is designed with the end-user in mind, promoting inclusivity and usability across different demographics and abilities.

### How the component works

This component will incorporate best practices from existing conversational AI interfaces. By analyzing applications such as ChatGPT and Hugging Face Chat, we can identify key UI elements and interaction patterns that contribute to a seamless user experience. The component will include features like intuitive input fields, responsive design, and clear feedback mechanisms to enhance user engagement and satisfaction.

### Real-world examples

https://github.com/huggingface/chat-ui

The generative AI initiative is in the innovation stages, but there is a growing trend to adopt AI-driven conversational interfaces. Examples from the private sector, such as ChatGPT and Hugging Face Chat, demonstrate the potential for such interfaces to revolutionize user interaction with digital services.

### When to use this component and when to consider something else

Use this component when creating AI agents for tasks such as advanced search, question-and-answer on websites, or interacting with documents such as PDF, text files, CSV, Microsoft 365 documents, and images. Consider alternative solutions if the project does not require real-time conversational interactions or if simpler, static interfaces are more appropriate for the user's needs.

### Usability considerations

The component must be versatile, functioning seamlessly across mobile, desktop, and web applications. It should be framework-independent, except for the USWDS implementation, to ensure broad compatibility and ease of integration. Usability testing will be essential to refine the interface and ensure it meets the needs of diverse users, including those with disabilities.

### Accessibility considerations

This component/s must be highly accessible, incorporating features such as text-to-speech and speech-to-text to support users with different abilities. Compliance with Section 508 standards ensures that all users, including those with disabilities, can interact effectively with the AI agents. The design should prioritize clarity, simplicity, and ease of use, with customizable settings to accommodate various user preferences and needs.

**Comment by HopeTurnerUSCIS** [comment]
One of the DHS websites, USCIS.gov, has a chatbot on the site today: https://www.uscis.gov/

We would like to learn how to improve this component so that we can submit it as a contribution to USWDS.
